builtin/providers/cloudstack/resource_cloudstack_loadbalancer_test.go

```diff
@@ -44,6 +45,14 @@ func TestAccCloudStackLoadBalancerRule_update(t *testing.T) {
 				Config: testAccCloudStackLoadBalancerRule_basic,
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudStackLoadBalancerRuleExist("cloudstack_loadbalancer_rule.foo"),
+					func(s *terraform.State) error {
+						rs, ok := s.RootModule().Resources["cloudstack_loadbalancer_rule.foo"]
+						if !ok {
+							return fmt.Errorf("Not found: cloudstack_loadbalancer_rule.foo")
+						}
+						id = rs.Primary.ID
+						return nil
+					},
```

Comment:
I'm specifically checking that the ID does not change after the update - I had a quick look around some other acceptance tests, but couldn't find an established pattern for this elsewhere.

`testAccCheckCloudStackLoadBalancerRuleExist` just checks that a rule exists, whereas i'm storing the ID after the creation (via the closure), and checking that it is the same after the update.